src/test/txvalidation_tests.cpp

```diff
@@ -59,5 +61,571 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_reject_coinbase, TestChainSetup)
+{
+    uint256 hash = SignatureHash(scriptPubKey, outTx, inIndex, SIGHASH_ALL, outTx.vout[outIndex].nValue, SigVersion::BASE);
+    signKey.Sign_Schnorr(hash, vchSig);
+    vchSig.push_back((unsigned char)SIGHASH_ALL);
+}
+
+BOOST_FIXTURE_TEST_CASE(tx_invalid_token_issue, TestChainSetup)
+{
+    const unsigned char vchKey[32] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0};
+    CKey key;
+    key.Set(vchKey, vchKey + 32, true);
+    CPubKey pubkey = key.GetPubKey();
+    std::vector<unsigned char> pubkeyHash(20);
+    std::vector<unsigned char> vchPubKey(pubkey.begin(), pubkey.end());
+    CHash160().Write(pubkey.data(), pubkey.size()).Finalize(pubkeyHash.data());
+
+    CMutableTransaction coinbaseSpendTx;
+    coinbaseSpendTx.nFeatures = 1;
+    coinbaseSpendTx.vin.resize(1);
+    coinbaseSpendTx.vout.resize(1);
+    coinbaseSpendTx.vin[0].prevout.hashMalFix = m_coinbase_txns[3]->GetHashMalFix();
+    coinbaseSpendTx.vin[0].prevout.n = 0;
+    coinbaseSpendTx.vout[0].nValue = 100 * CENT;
+    coinbaseSpendTx.vout[0].scriptPubKey = CScript() << OP_DUP << OP_HASH160 << ToByteVector(pubkeyHash) << OP_EQUALVERIFY << OP_CHECKSIG;
+
+    std::vector<unsigned char> vchSig;
+    CMutableTransaction coinbaseIn(*m_coinbase_txns[3]);
+    Sign(vchSig, coinbaseKey, m_coinbase_txns[3]->vout[0].scriptPubKey, coinbaseIn, 0, coinbaseSpendTx, 0);
+    coinbaseSpendTx.vin[0].scriptSig = CScript() << vchSig;
+
+    testTx(this, MakeTransactionRef(coinbaseSpendTx), true);
+
+    CScript scriptPubKey = CScript() << ColorIdentifier().toVector() << OP_COLOR << OP_DUP << OP_HASH160 << ToByteVector(pubkeyHash) << OP_EQUALVERIFY << OP_CHECKSIG;
+    CMutableTransaction tokenIssueTx;
+
+    tokenIssueTx.nFeatures = 1;
+    tokenIssueTx.vin.resize(1);
+    tokenIssueTx.vout.resize(1);
+    tokenIssueTx.vin[0].prevout.hashMalFix = coinbaseSpendTx.GetHashMalFix();
+    tokenIssueTx.vin[0].prevout.n = 0;
+    tokenIssueTx.vout[0].nValue = 100 * CENT;
+    tokenIssueTx.vout[0].scriptPubKey = scriptPubKey;
+
+    Sign(vchSig, key, coinbaseSpendTx.vout[0].scriptPubKey, coinbaseSpendTx, 0, tokenIssueTx, 0);
+    std::vector<unsigned char> vchPubKey0(pubkey0.begin(), pubkey0.end());
+    tokenIssueTx.vin[0].scriptSig = CScript() << vchSig << vchPubKey;
+
+    testTx(this, MakeTransactionRef(tokenIssueTx), false, "invalid-colorid");
+}
+
+/*
+Test token type REISSUABLE
+Txs:
+coinbaseSpendTx
+tokenIssueTx(from coinbaseSpendTx) - 100 tokens
+tokenTransferTx - 1. no fee
+                - 2. split into 50 + 40 tokens - token balance error
+                - 3. success
+test transaction reissue
+tokenAggregateTx- 1. no fee
+                - 2. add extra tokens - token balance error
+                - 3. success
+TODO: tokenBurnTx     - 1. no fee
+                - 2. add extra TPC - token balance error
```

Comment:
Will this TODO be resolved with another pull request?

Reply:
I initially thought I can create another PR. Now I have finished testing token burn cases. So I'll just update this PR